bugalink-backend/trips/views.py

```diff
@@ -81,3 +88,46 @@ def reject(self, request, *args, **kwargs):
         trip_request.status = "REJECTED"
         trip_request.save()
         return Response(self.get_serializer(trip_request).data)
+
+
+class TripRecommendationViewSet(
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = Trip.objects.all()
+    serializer_class = TripSerializer
+
+    @action(detail=True, methods=["get"])
+    def get(self, request, user_id, *args, **kwargs):
+        try:
+            data = {"trips": []}
+            for passenger_routine in PassengerRoutine.objects.filter(
+                passenger__user_id=user_id
+            ):
+                data["trips"] = (
+                    data["trips"]
+                    + PassengerRoutineSerializer(passenger_routine)
+                    .get_recommendations(obj=passenger_routine)
+                    .data
+                )
+
+            return Response(data, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"msg": str(e)}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Here you call the error message "msg" whereas a few lines down you call it "message". This makes it impossible in the frontend to extract the error messages from the request to show them to the user because they will have a different name depending on the endpoint